### PR TITLE
fix: replace hardcoded /tmp paths with secure tempfile.mkdtemp()

### PR DESCRIPTION
**Pull request by 1PoPTRoN**
# Type of change

- [ ] Refactor
- [ ] New feature
- [x] Bug fix
- [ ] Optimization

# Description

Hardcoded `/tmp` paths (`rollback_versions_directory: /tmp/kraken-rollback` and `archive_path: /tmp`) are world-writable, exposing krkn to TOCTOU race conditions and symlink attacks. 

This PR replaces them with `tempfile.mkdtemp()` which atomically creates directories with unpredictable names and `0o700` (owner-only) permissions, eliminating both attack vectors. Users can still set explicit paths in config if preferred.

**Changes:**
- `run_kraken.py`: Secure fallback via `tempfile.mkdtemp()` when paths are empty/unset
- All config files: Removed hardcoded `/tmp` defaults
- `tests/test_rollback.py`: Test dirs use `tempfile.mkdtemp()`, fixed `TestScenarioPlugin.run()` signature to match `AbstractScenarioPlugin.run()`, added `TestSecureTempDirectories` test class
- Also added `skipif` for `test_simple_rollback_scenario_plugin` which requires a running Kubernetes cluster (integration test). It runs in CI where a kind cluster is available, and skips gracefully on dev machines without a kubeconfig.

# Documentation
- [ ] **Is documentation needed for this update?**

# Checklist before requesting a review
- [x] Ensure the changes and proposed solution have been discussed in the relevant issue and have received acknowledgment from the community or maintainers. See [contributing guidelines](https://krkn-chaos.dev/docs/contribution-guidelines/)
- [x] I have performed a self-review of my code by running krkn and specific scenario
- [x] If it is a core feature, I have added thorough unit tests with above 80% coverage

*REQUIRED*:
Description of combination of tests performed and output of run

```bash
# With kind cluster (26/26 passed)
python -m pytest tests/test_rollback.py -v
...
```
Screenshot 2026-04-02 at 7 08 34 AM

### CODE REVIEW

File: run_kraken.py [hunk 1]
```diff
@@ -88,17 +89,21 @@ def main(options, command: Optional[str]) -> int:
...
                 "auto_rollback",
                 False
             ),
-            versions_directory=get_yaml_item_value(
-                config["kraken"],
-                "rollback_versions_directory",
-                "/tmp/kraken-rollback"
-            ),
+            versions_directory=rollback_versions_dir,
         )
```
[the remark this answers is not in the document]
Comment (1PoPTRoN) [reply run_kraken.py:115]: Fixed, now defaults to `~/.krkn/rollback`

File: run_kraken.py [hunk 2]
```diff
@@ -88,17 +89,21 @@ def main(options, command: Optional[str]) -> int:
...
         port = get_yaml_item_value(config["kraken"], "port", 8081)
+        rollback_versions_dir = get_yaml_item_value(
+            config["kraken"],
+            "rollback_versions_directory",
+            ""
+        )
+        if not rollback_versions_dir:
+            rollback_versions_dir = tempfile.mkdtemp(prefix="krkn-rollback-")
+            logging.info("Using secure temp directory for rollback versions: %s" % rollback_versions_dir)
         RollbackConfig.register(
```
[the remark this answers is not in the document]
Comment (1PoPTRoN) [reply run_kraken.py:108]: Fixed, archive dirs are cleaned up via `atexit.register`

File: tests/test_rollback.py [hunk 3]
```diff
@@ -15,8 +16,8 @@
 sys.path.append(
     os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 )  # Adjust path to include krkn
-TEST_LOGS_DIR = "/tmp/krkn_test_rollback_logs_directory"
-TEST_VERSIONS_DIR = "/tmp/krkn_test_rollback_versions_directory"
+TEST_LOGS_DIR = tempfile.mkdtemp(prefix="krkn_test_rollback_logs_")
+TEST_VERSIONS_DIR = tempfile.mkdtemp(prefix="krkn_test_rollback_versions_")
 
 
 class TestRollbackScenarioPlugin:
```
[the remark this answers is not in the document]
Comment (1PoPTRoN) [reply tests/test_rollback.py:23]: These are small temp dirs under /tmp, cleaned up by the OS. `tmp_path_factory` adds complexity without meaningful benefit here.

File: tests/test_rollback.py [hunk 4]
```diff
@@ -128,6 +129,10 @@ def setup_rollback_config(self):
         )
 
     @pytest.mark.usefixtures("setup_rollback_config")
+    @pytest.mark.skipif(
+        not os.path.isfile(os.path.expanduser(os.getenv("KUBECONFIG", "~/.kube/config"))),
```
[the remark this answers is not in the document]
Comment (1PoPTRoN) [reply tests/test_rollback.py:133]: `KUBECONFIG` is set as a single path in krkn's CI environment. This edge case is out of scope for this fix.
